<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2016-09-08 Thu 08:29 -->
<meta  http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta  name="viewport" content="width=device-width, initial-scale=1" />
<title>RNA sequence analysis with reference genome</title>
<meta  name="generator" content="Org-mode" />
<meta  name="author" content="Thomas Källman" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: visible;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  pre.src-sh:before    { content: 'sh'; }
  pre.src-bash:before  { content: 'sh'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-R:before     { content: 'R'; }
  pre.src-perl:before  { content: 'Perl'; }
  pre.src-java:before  { content: 'Java'; }
  pre.src-sql:before   { content: 'SQL'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  /*]]>*/-->
</style>
<script type="text/javascript">
/*
@licstart  The following is the entire license notice for the
JavaScript code in this tag.

Copyright (C) 2012-2013 Free Software Foundation, Inc.

The JavaScript code in this tag is free software: you can
redistribute it and/or modify it under the terms of the GNU
General Public License (GNU GPL) as published by the Free Software
Foundation, either version 3 of the License, or (at your option)
any later version.  The code is distributed WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  See the GNU GPL for more details.

As additional permission under GNU GPL version 3 section 7, you
may distribute non-source (e.g., minimized or compacted) forms of
that code without the copy of the GNU GPL normally required by
section 4, provided you include this license notice and a URL
through which recipients can access the Corresponding Source.


@licend  The above is the entire license notice
for the JavaScript code in this tag.
*/
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
</head>
<body>
<div id="content">
<h1 class="title">RNA sequence analysis with reference genome</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#orgheadline1">1. Introductionoo and data description</a></li>
<li><a href="#orgheadline2">2. Uppmax</a></li>
<li><a href="#orgheadline3">3. Create a folder structure for and collect data for analysis</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgheadline1" class="outline-2">
<h2 id="orgheadline1"><span class="section-number-2">1</span> Introductionoo and data description</h2>
</div>

<div id="outline-container-orgheadline2" class="outline-2">
<h2 id="orgheadline2"><span class="section-number-2">2</span> Uppmax</h2>
<div class="outline-text-2" id="text-2">
<p>
All the tools needed for analysis of this data set are available as
modules on uppmax and hence need to be loaded to be used. Just as earlier
this week modules are loaded with commands like this one:
</p>

<div class="org-src-container">

<pre class="src src-bash">module load bioinfo-tools
module load FastQC/0.11.5
</pre>
</div>

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left">RNAseq-practical.html</td>
</tr>

<tr>
<td class="org-left">RNAseq-practical.org</td>
</tr>

<tr>
<td class="org-left">ditaa-seqboxes.png</td>
</tr>
</tbody>
</table>


<p>
Also remember that this need to be done on the node where you want to
do the analysis if you already have an active reservation you have to
ssh to your reserved node. If you are unsure weather you have already
made a reservation you can look for all the jobs your user is
currently running with the following command (remember to change
username to your own uppmax username).
</p>

<div class="org-src-container">

<pre class="src src-sh">jobinfo -u username
</pre>
</div>

<p>
In the list of active jobs you can see the name of the compute node
that your reservation is associated with. Usually they are named M1,
M2. Once you hav the name you can use the following command to go to
that node. 
</p>

<div class="org-src-container">

<pre class="src src-sh">ssh M1
</pre>
</div>

<p>
If you do not have any job running, you need to ask for a new job using
the course reservation. 
</p>

<div class="org-src-container">

<pre class="src src-sh">salloc -A g2016001 -t 08:00:00 -p core -n 8 --no-shell --reservation=g2016008_4 &amp;
</pre>
</div>

<p>
Please make sure that you do not have multiple jobs running at once as
that means that other course members will not be able to get access to
compute nodes. 
</p>
</div>
</div>

<div id="outline-container-orgheadline3" class="outline-2">
<h2 id="orgheadline3"><span class="section-number-2">3</span> Create a folder structure for and collect data for analysis</h2>
<div class="outline-text-2" id="text-3">
<p>
At this time point you should be logged in to your compute node and
ready to start the analysis. If you are unsure weather you are really
at the compute node you can check the shell prompt that you
terminal is displaying. If it says username$milou1 or
username$milou2 your are still on the login node so make sure that it
instead says something like username$m1.
</p>

<p>
Before we start the analysis we will set up a suitable folder
structure for our data and results. This facilitates analysis and
makes it easy to trace what has been done and locate files of
interest.  How one best structure this differs from person to person,
but there are common patterns that most people tend to follow.
</p>

<p>
First of all one start at a place that have enough space to hold your
data and analysis and decide on a folder, where one decides to store
the data. For this class we suggest you create a folder for the raw
reads. One folder for the results of mapping, one for genome sequence
and annotations and finally one for the Expression analysis.
</p>

<p>
Go to the place where you want to store you data and analysis and
create four folders with the following commands:
</p>

<div class="org-src-container">

<pre class="src src-sh:">mkdir Reads
mkdir Bams
mkdir Expression
mkdir Reference
</pre>
</div>

<p>
The final step before we start analysing data is to collect the files
needed for analysis. If you are at the main project folder the
following commands will create copies of the files needed for the
analysis to your workspace.
</p>

<div class="org-src-container">

<pre class="src src-sh">cp path_to_reads/* Reads/
cp path_to_refs/* Reference/
</pre>
</div>

<p>
If you have managed to go through these steps you should now have the
following structure of files and folder in your current work directory
</p>


<div class="figure">
<p><img src="ditaa-seqboxes.png" alt="ditaa-seqboxes.png" />
</p>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="date">Date: September, 2016</p>
<p class="author">Author: Thomas Källman</p>
<p class="date">Created: 2016-09-08 Thu 08:29</p>
<p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
